packages/analytics/get-product-intent-from-url/index.ts

```diff
@@ -0,0 +1,44 @@
+const products = [
+  'packer',
+  'terraform',
+  'vault',
+  'boundary',
+  'consul',
+  'nomad',
+  'waypoint',
+  'vagrant',
+] as const
+
+type Products = typeof products[number]
+
+export const getProductIntentFromURL = (url: string): Products | null => {
```

Comment:
Went back and forth on this utility just reading window.location vs accepting a URL. Curious to hear if that is a simpler approach?

Reply:
💭 It could optionally accept a URL, but fall back to reading from `window` if no url is specified.

Reply:
Updated here https://github.com/hashicorp/web-platform-packages/pull/63/commits/45a37a3e542961ba065295cc60fa18eb643a734d